src/include/platform/internal/GenericConfigurationManagerImpl.cpp

```diff
@@ -248,9 +248,14 @@ CHIP_ERROR GenericConfigurationManagerImpl<ConfigClass>::StoreManufacturingDate(
 
 template <class ConfigClass>
 void GenericConfigurationManagerImpl<ConfigClass>::InitiateFactoryReset()
+{}
+
+template <class ImplClass>
+void GenericConfigurationManagerImpl<ImplClass>::StartAdvertisement()
 {
 #if CHIP_ENABLE_ROTATING_DEVICE_ID
-    IncrementLifetimeCounter();
+    // Increment life time counter to protect against long-term tracking of rotating device ID.
+    _IncrementLifetimeCounter();
```

Comment:
comment below should probably be inside the #if ... if no rotating device id, there is no lifetime counter.